src/create_atoms.cpp

```diff
@@ -800,25 +805,54 @@ void CreateAtoms::add_random()
       // check for overlap of new atom/mol with all other atoms
       //   including prior insertions
       // minimum_image() needed to account for distances across PBC
-      // new molecule only checks its center pt against others
-      //   odistsq is expanded for mode=MOLECULE to account for molecule size
 
       if (overlapflag) {
         double **x = atom->x;
         int nlocal = atom->nlocal;
 
         int reject = 0;
-        for (int i = 0; i < nlocal; i++) {
-          delx = xone[0] - x[i][0];
-          dely = xone[1] - x[i][1];
-          delz = xone[2] - x[i][2];
-          domain->minimum_image(delx, dely, delz);
-          distsq = delx * delx + dely * dely + delz * delz;
-          if (distsq < odistsq) {
-            reject = 1;
-            break;
+        if (mode == ATOM) {
+          for (int i = 0; i < nlocal; i++) {
+            delx = xone[0] - x[i][0];
+            dely = xone[1] - x[i][1];
+            delz = xone[2] - x[i][2];
+            domain->minimum_image(delx, dely, delz);
+            distsq = delx * delx + dely * dely + delz * delz;
+            if (distsq < odistsq) {
+              reject = 1;
+              break;
+            }
+          }
+        } else {
+          int incr;
+          if (comm->me == 0) {
```

Comment:
I don't understand why one proc creates the molecule, then Bcasts its coords to the others.  Similar to the random generation of xone above, won't all procs create the same randomly oriented molecule?

Reply:
I was probably worried that it is possible that different processors could call get_xmol a different number of times, resulting in ranmol->uniform getting out of sync.

Reply:
ah, I see that get_xmol() uses ranmol and not random
Which means doing it on proc 0 followed by a Bcast is needed

I don't think xmolbuf and the double copy is needed. You can just do
MPI_Bcast(&xmol[0][0],onemol->natoms*3,...)

Reply:
I think I was worried that this array was not guaranteed to be contiguous. But, looking at memory.h it looks like we should be okay. Still, just want to confirm that you agree before making the change

Reply:
if you mean is the 3*n data contiguous starting in xmol at location xmol[0][0], then yes it is
That is true for all multidimensional arrays allocated by the Memory class.
That was actually one of the main motivations for the Memory class.